### PR TITLE
Some changes add third step of add product

### PR DESCRIPTION
**Pull request by djanbolotov**
Changes: 
Added third step of add product page

Снимок экрана 2024-07-11 в 20 36 01

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [ ] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet
### Frontend
- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/libs/types/categories.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+type Category = {
```
Comment (VladManukian) [inline src/libs/types/categories.ts:1]: Does we need both type Category of them now? 
https://github.com/ZenBit-Tech/sunny-bunnies_fe/blob/develop/src/libs/types/products.ts

File: src/pages/add-product/components/first-step/index.tsx [hunk 2]
```diff
@@ -20,53 +21,63 @@ import { ImageList } from "../image-list/index.tsx";
...
-	selected: boolean;
-	src: string;
-};
-
-type FormData = {
-	images: Image[];
-};
-
-const ImageUpload: React.FC = () => {
+interface ImageUploadProps {
```
Comment (VladManukian) [inline src/pages/add-product/components/first-step/index.tsx]:
```suggestion
type ImageUploadProps {
```

File: src/pages/add-product/hooks/useProductData.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+			images: images,
+		}));
+	};
+
+	const setSecondStepData = (
+		formData: SecondStepFormData,
+		categories: Category[],
+	): void => {
+		const { category, style, type } = formData;
+		const categoryObject = categories.find((c) => c.name === category);
```
Comment (VladManukian) [inline src/pages/add-product/hooks/useProductData.ts]: @lyudad 
` categories.find((c) => c.name === category);`
I'm interested in your opinion.
Do you use such abbreviations on projects or not?)

File: src/libs/types/categories.ts (hunk 1 above)
Comment (djanbolotov) [reply src/libs/types/categories.ts:1]: Yeah you'r right i'll remove one of them

File: src/pages/add-product/components/second-step/index.tsx [hunk 4]
```diff
@@ -110,22 +129,24 @@ const SecondStepForm: React.FC = () => {
...
 	}, [selectedCategory, selectedType, selectedStyle, setValue, getValues]);
 
-	const onSubmit = (data: FormData): void => {
-		alert(data);
+	const onSubmit = (data: SecondStepFormData): void => {
+		if (categories) setSecondStepData(data, categories);
 		navigate(AppRoute.PRODUCT_DESCRIPTION);
 	};
 
+	if (!categories) return <div>Loading</div>;
```
Comment (lyudad) [inline src/pages/add-product/components/second-step/index.tsx]: 1.3.3

File: src/pages/add-product/hooks/useProductData.ts (hunk 3 above)
Comment (lyudad) [reply src/pages/add-product/hooks/useProductData.ts]: no, no abreviation, code is our documentation

Comment (lyudad) [reply src/pages/add-product/hooks/useProductData.ts]: calegotyElement as option

File: src/pages/add-product/components/form-input-field/index.tsx
Comment (lyudad) [inline src/pages/add-product/components/form-input-field/index.tsx:1]: input better to make reusable component and move it to components order in root

File: src/pages/add-product/hooks/useProductData.ts
Comment (lyudad) [inline src/pages/add-product/hooks/useProductData.ts:1]: as this hooks connect to only one component, better to have them in same folder that component is

File: src/pages/add-product/validation/first-step-validation.ts
Comment (lyudad) [inline src/pages/add-product/validation/first-step-validation.ts:1]: as it connect to one component better to have it near component

**Pull request merged**